components/suggestion/suggestion_box.jsx

```diff
@@ -343,7 +343,7 @@ export default class SuggestionBox extends React.PureComponent {
 
     handleChange = (e) => {
         const textbox = this.getTextbox();
-        const pretext = textbox.value.substring(0, textbox.selectionEnd).toLowerCase();
```

Comment:
(3/5) Since we're no longer lower casing automatically, are there any other components relying on this still expecting it to be lower cased?

Reply:
@larkox Do you recall why this change was needed here? https://github.com/mattermost/mattermost-webapp/pull/7555/files#diff-76b8f06eaf4a39bfd40b9bd99d74c9b9d7beb4e56a9efb19fd1848a6955ee911

Reply:
If I remember correctly, this was done to handle better the casing, and make sure we respect user provided casing when autocompleting (e.g. if I write `/JIRA create t` and I select the command `ticket` the user expects only the ticket part being updated and showing `/JIRA create ticket`. Previously it would have changed everything and leave something like `/jira create ticket`).

After a quick check:
- At mention already is lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/at_mention_provider/at_mention_provider.jsx#L302
- Channel mention provider is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/channel_mention_provider.jsx#L68
- Emoticon provider may present some problems, we have to investigate: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/emoticon_provider.jsx#L63
- Generic channel provider is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/generic_channel_provider.jsx#L56
- Generic user provider is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/generic_user_provider.jsx#L69
- Search channel provider is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/search_channel_provider.jsx#L37
- Search date is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/search_date_provider.jsx#L9
- Search user is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/search_user_provider.jsx#L72
- Handle mobile for commands is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/command_provider/command_provider.tsx#L125
- Handle webapp for commands is already lowercasing: https://github.com/mattermost/mattermost-webapp/blob/99e04158c01d5741d19b7dc228751af9972a6825/components/suggestion/command_provider/command_provider.tsx#L171

Reply:
I just confirmed that emoticons will no longer show suggestion when mixed with uppercase. Will submit a quick fix.

Reply:
https://github.com/mattermost/mattermost-webapp/pull/7822